theano/gof/cc.py

```diff
@@ -1565,16 +1689,39 @@ def get_dynamic_module(self):
         """ % dict(struct_name=self.struct_name)
 
         # We add all the support code, compile args, headers and libs we need.
-            for support_code in self.support_code() + self.c_support_code_apply:
-                mod.add_support_code(support_code)
-            mod.add_support_code(self.struct_code)
-            mod.add_support_code(static)
-            mod.add_function(instantiate)
-            for header in self.headers():
-                mod.add_include(header)
-            for init_code_block in self.init_code() + self.c_init_code_apply:
-                mod.add_init_code(init_code_block)
-            self._mod = mod
+                for support_code in self.support_code() + self.c_support_code_apply:
```

Comment:
I think all the new code in this section should be less indented by 1 level.

I think this should fix problems in the tests.